Q1. Descriptive analysis of data is essential for business analytics

(A) Please write a Python code to read all postal codes and save them in a list.

In [21]:
file = open('postalcode.txt', 'r')

postal_code = []

for line in file.readlines():
    post_cod = line.strip().split(',')
    postal_code.append(post_cod)
print(postal_code)

[['Bordeaux', '33000'], ['Ambares-et-Lagrave', '33440'], ['Ambes', '33810'], ['Artigues-pres-Bordeaux', '33370'], ['Brassens', '33530'], ['Begles', '33130'], ['Blanquefort', '33290'], ['Bruges', '33520'], ['Carbon-Blanc', '33560'], ['Cenon', '33150'], ['Eysines', '33320'], ['Floriac', '33270'], ['Gradignan', '33170'], ['Le Bouscat', '33110'], ['Le Haillan', '33185'], ['Lormont', '33310'], ['Martignas-sur-Jalle', '33127'], ['Merignac', '33700'], ['Pessac', '33600'], ['Saint-Aubin-de-Medoc', '33160'], ['Talence', '33400'], ['Ornon', '33140']]


(B) Continue with (A). Please export descriptive statistics of price and size of apartments (i.e., mean, std, min, max, median, Q25, Q75) for every postal code (that you read in A) as a csv file (e.g., 33110.csv, 33440.csv and so on) in Bordeaux Metropole.

In [22]:
# Load libraries
import pandas as pd

In [23]:
df_housing = pd.read_csv('Bordeaux_Metropole.csv')

df_housing_appartment = df_housing.query('type_local == "Appartement"')
print(df_housing_appartment)

      date_mutation  valeur_fonciere  code_postal section   type_local  \
0          2014/2/6         155000.0        33000      KX  Appartement   
3        2014/10/31         200000.0        33000      HW  Appartement   
4        2014/10/31         200000.0        33000      HW  Appartement   
6         2015/9/25         212000.0        33000      IW  Appartement   
8         2015/10/9         187000.0        33000      HD  Appartement   
...             ...              ...          ...     ...          ...   
76409     2015/2/26         256000.0        33140      AZ  Appartement   
76412     2017/3/29         307000.0        33140      BS  Appartement   
76413     2017/3/29         307000.0        33140      BS  Appartement   
76428     2014/7/31          83000.0        33140      AB  Appartement   
76432     2017/2/14         155000.0        33140      AB  Appartement   

       surface_relle_bati  nombre_pieces_principales  surface_terrain  
0                      32              

In [24]:
df_group = df_housing_appartment.groupby(['code_postal'])
df_group.describe()

valeur_fonciere                                                  \
                      count          mean           std      min        25%   
code_postal                                                                   
33000               14235.0  4.534332e+05  8.672420e+05      1.0  139500.00   
33110                1648.0  4.924541e+05  1.284856e+06      1.0  134601.25   
33127                 393.0  1.990184e+05  1.875097e+05      1.0  137800.00   
33130                2466.0  5.743806e+05  1.619805e+06      1.0  135510.00   
33140                3063.0  6.378683e+05  1.634815e+06  30000.0  149500.00   
33150                2311.0  4.011639e+05  1.034071e+06      0.0  120176.50   
33160                 605.0  9.650681e+05  2.263725e+06   2100.0  156000.00   
33170                1368.0  7.785822e+05  1.846409e+06   1000.0   90950.00   
33185                 356.0  6.297428e+05  1.613210e+06  75000.0  157374.75   
33270                 589.0  1.987774e+05  1.474968e+05  25000.0  131500.00   
33290                 889.0  7.676672e+05  1.754283e+06  30990.0  128500.00   
33310                1236.0  4.287376e+05  9.568946e+05      0.0  115665.00   
33320                1208.0  3.709477e+05  7.634388e+05  39480.0  144449.95   
33370                 544.0  4.133069e+05  1.086696e+06  29190.0  139382.50   
33400                3921.0  1.046425e+06  2.378162e+06      5.0  109533.00   
33440                 267.0  3.745676e+05  4.031917e+05  55100.0  114700.00   
33520                1472.0  2.290274e+05  2.696030e+05      0.0  147000.00   
33530                 215.0  1.997768e+05  1.279435e+05  73300.0  151254.00   
33560                 309.0  1.930805e+05  1.648389e+05  30000.0  117670.00   
33600                2761.0  3.468511e+05  6.934810e+05    250.0  122860.00   
33700                5820.0  4.321404e+05  1.597139e+06      1.0  129000.00   
33810                  82.0  2.677392e+05  3.097934e+05  60000.0  132875.00   

                                                 surface_relle_bati  \
                   50%          75%          max              count   
code_postal                                                           
33000        236050.00  440000.0000  10789484.00            14235.0   
33110        174175.00  250625.0000   6662829.00             1648.0   
33127        161215.38  202500.0000   1203914.73              393.0   
33130        170673.50  225725.0000   9309300.00             2466.0   
33140        176567.00  235900.0000   9975263.00             3063.0   
33150        155000.00  206000.0000   8115000.00             2311.0   
33160        200000.00  258550.0000   7894646.50              605.0   
33170        150000.00  231718.1425   9050000.00             1368.0   
33185        183500.00  230000.0000   7894646.50              356.0   
33270        163500.00  205000.0000   1360000.00              589.0   
33290        163500.00  222050.0000   6403637.50              889.0   
33310        144000.00  184625.0000   3921037.00             1236.0   
33320        175000.00  223525.0000   6231966.00             1208.0   
33370        161853.00  207500.0000   7485710.00              544.0   
33400        166600.00  297000.0000  10500000.00             3921.0   
33440        168561.87  415000.0000   1258446.20              267.0   
33520        178000.00  220000.0000   2944440.00             1472.0   
33530        165426.42  207450.0000    730437.00              215.0   
33560        135000.00  186604.0000    935000.00              309.0   
33600        168000.00  242000.0000   4825000.00             2761.0   
33700        166000.00  220500.0000  16142400.00             5820.0   
33810        145000.00  188012.5000   1040100.00               82.0   

                        ... nombre_pieces_principales       surface_terrain  \
                  mean  ...                       75%   max           count   
code_postal             ...                                                   
33000        55.402669 

Q2: Data analysis and Visualization